src/ui/public/vislib/visualizations/gauges/simple.js

```diff
@@ -0,0 +1,208 @@
+
+export function SimpleGaugeProvider() {
+
+
+  const defaultConfig = {
+    showTooltip: true,
+    percentageMode: true,
+    extents: [0, 10000],
+    scale: {
+      show: true,
+      color: '#666',
+      width: 2,
+      ticks: 10,
+      tickLength: 5,
+    },
+    labels: {
+      show: true,
+      color: '#666'
+    },
+    style: {
+      bgColor: true,
+      bgFill: '#666'
+    }
+  };
+
+  class SimpleGauge {
+    constructor(gaugeChart) {
+      this.gaugeChart = gaugeChart;
+      this.gaugeConfig = gaugeChart.gaugeConfig;
+      this.gaugeConfig = _.defaultsDeep(this.gaugeConfig, defaultConfig);
+      this.randomNumber = Math.round(Math.random() * 100000);
+
+      this.gaugeChart.handler.visConfig.set('legend', {
+        labels: this.getLabels(),
+        colors: this.getColors()
+      });
+
+      const colors = this.gaugeChart.handler.visConfig.get('legend.colors', null);
+      if (colors) {
+        this.gaugeChart.handler.vis.uiState.setSilent('vis.defaultColors', null);
+        this.gaugeChart.handler.vis.uiState.setSilent('vis.defaultColors', colors);
+      }
+
+      this.colorFunc = this.gaugeChart.handler.data.getColorFunc();
+    }
+
+    getLabels() {
+      const isPercentageMode = this.gaugeConfig.percentageMode;
+      const colorsRange = this.gaugeConfig.colorsRange;
+      const max = _.last(colorsRange).to;
+      const labels = [];
+      colorsRange.forEach(range => {
+        const from = isPercentageMode ? Math.round(100 * range.from / max) : range.from;
+        const to = isPercentageMode ? Math.round(100 * range.to / max) : range.to;
+        labels.push(`${from} - ${to}`);
+      });
+
+      return labels;
+    }
+
+    getColors() {
+      const invertColors = this.gaugeConfig.invertColors;
+      const colorSchema = this.gaugeConfig.colorSchema;
+      const colorsRange = this.gaugeConfig.colorsRange;
+      const labels = this.getLabels();
+      const colors = {};
+      for (const i in labels) {
+        if (labels[i]) {
+          const val = invertColors ? 1 - i / colorsRange.length : i / colorsRange.length;
+          colors[labels[i]] = getHeatmapColors(val, colorSchema);
+        }
+      }
+      return colors;
+    }
+
+    getBucket(val) {
+      let bucket = _.findIndex(this.gaugeConfig.colorsRange, range => {
+        return range.from <= val && range.to > val;
+      });
+
+      if (bucket === -1) {
+        if (val < this.gaugeConfig.colorsRange[0].from) bucket = 0;
+        else bucket = this.gaugeConfig.colorsRange.length - 1;
+      }
+
+      return bucket;
+    }
+
+    getLabel(val) {
+      const bucket = this.getBucket(val);
+      const labels = this.gaugeChart.handler.visConfig.get('legend.labels');
+      return labels[bucket];
+    }
+
+    getColorBucket(val) {
+      const bucket = this.getBucket(val);
+      const labels = this.gaugeChart.handler.visConfig.get('legend.labels');
+      return this.colorFunc(labels[bucket]);
```

Comment:
The color-mapping seems to miss the last color of the defined range. 

E.g., the green-red color range should end with a red color, instead of with an orange.

![image](https://cloud.githubusercontent.com/assets/1833023/25806721/26c13c9c-33d2-11e7-8009-a99df5311e9b.png)

We have the same issue when inverting. The range should begin with red and end with a dark-green color.

![image](https://cloud.githubusercontent.com/assets/1833023/25806760/4a29684e-33d2-11e7-8c17-d5e0dec7c8dc.png)